account-info/src/urls.rs

```diff
@@ -1,3 +1,44 @@
+use contract::{contract_api::{runtime, storage}, unwrap_or_revert::UnwrapOrRevert};
+use types::{Key, URef, account::AccountHash};
+use super::ContractError;
+
+pub const URLS_DICT: &str = "urls";
+
 pub struct Urls {
     dict_uref: URef
+}
+
+impl Urls {
+    pub fn new() -> Urls {
+        let dict_key: Key = runtime::get_key(URLS_DICT).unwrap_or_revert();
+        let dict_uref: &URef = dict_key.as_uref().unwrap_or_revert();
+        Urls {
+            dict_uref: *dict_uref
+        }
+    }
+
+    pub fn set(&self, address: &AccountHash, url: &str) {
+        if !is_valid_url(url) {
+            runtime::revert(ContractError::BadUrlFormat);
+        }
+        storage::dictionary_put(self.dict_uref, &address.to_string(), url);
+    }
+    
+    pub fn delete(&self, address: &AccountHash) {
+        storage::dictionary_put(self.dict_uref, &address.to_string(), "");
```

Comment:
Shouldn't we delete the key here instead of setting the value to an empty string?

Reply:
Right now there is no function that deletes a key from a dictionary.

Reply:
Got it!